.github/workflows/check.yml

```diff
@@ -18,6 +18,21 @@ jobs:
           mdl -s .github/linter/readme_style.rb README.md
           mdl -s .github/linter/base_style.rb .github
 
+  cairo-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Rust nightly
+        uses: dtolnay/rust-toolchain@nightly
+        with:
+          components: rustfmt, clippy
+      - name: Install cairo-lint
+        run: cargo install scarb-cairo-lint --git https://github.com/keep-starknet-strange/cairo-lint
+
+      - uses: asdf-vm/actions/install@v3
+      - run: scarb cairo-lint
```

Comment:
I was thinking the CI check should fail if there are existing Cairo-lint warnings. What do you think?

Reply:
I am not sure about this, there are some linting warning I don't think we might want to follow.

Example: 
```
warning: Plugin diagnostic: Leaving `panic!` in the code is discouraged.
  --> /home/runner/work/shinigami/shinigami/packages/engine/src/cond_stack.cairo:90:18
   |
90 |             _ => panic!("Invalid condition")
```

Reply:
I'll see if there is a way to ignore lints, but I think all the existing lints we want to follow. Even the panic one, we don't want any panic conditions ( and this branch should never be reached anyway ).